CHANGELOG.md

```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+### Added
```

Comment:
This is not new feature, but bugfix (DateTime is already supported, however expects that everything is in UTC). 

Pls change from "Added" to "Fixed" section, as we are using https://keepachangelog.com/en/1.0.0/

Reply:
> Seems that [current_timezone()](https://github.com/SAP/python-pyodata/blob/21a99b50c3b1718e5b83109d6b4d944d06b7d1e5/pyodata/v2/model.py#L31) method (which should probably be deleted in the PR as well, since this replaces its usages) and the comments there are wrong.

Not sure I understand you correctly, but this would essentially mean that Edm.DateTime will be converted to [naive datetime](https://docs.python.org/3/library/datetime.html) objects, instead of "aware" ones with a hard coded UTC timezone?

> But then, this same problem should be fixed for Edm.DateTime as well, not only for Edm.DateTimeOffset. Correct?

The literal form of Edm.DateTime in (Overview)[https://www.odata.org/documentation/odata-version-2-0/overview/#AbstractTypeSystem) does not mention either, timezone nor offset. [Atom Format](https://www.odata.org/documentation/odata-version-2-0/atom-format/) references the literal form directly, therefore can also not represent a timezone/offset. [JSON](https://www.odata.org/documentation/odata-version-2-0/json-format/) however has an offset described. :confused:

As I have yet not seen a timezone/offset on a Edm.DateTime, I assumed it will never have an offset. Additionally, what would be the purpose of Edm.DateTimeOffset if both types can contain a timezone?

Possible way forward:
- Edm.DateTime: Remove fake timezone information (separate PR?)
- Edm.DateTimeOffset: Add timezone information (original idea of this PR)

Reply:
> This is not new feature, but bugfix (DateTime and DateTimeOffset are already supported, however they expects that everything incoming is in UTC).

The level of support currently differs:
- DateTime: Converted to a Python datetime object with fake/guessed UTC timezone
- DateTimeOffset: Converted to Python string. Works well enough to fetch and send to services, but not when one wants to perform work on it. This PR converts to/from a Python datetime object with the proper timezone attached.

> Pls change from "Added" to "Fixed" section, as we are using https://keepachangelog.com/en/1.0.0/

Will do.

Reply:
Edm.DateTime - while odata.org has is just datetime’yyyy-mm-ddThh:mm[:ss[.fffffff]]’ - and SAP Gateway returns this in UTC by design, so it is understandable that pyodata started from there - the MS ODATA specification for odata has optional timezone as well for this. That's why I thought that what you did for Edm.DateTimeOffset type should be applied to Edm.DateTime as well - and apply UTC as the default, if no infomration about timezone is received. 

                 
```
dateTimeLiteral =   year "-" 
                    month "-" 
                    day "T"
                    hour ":" 
                    minute 
                    [":" second ["." nanoSeconds]] 
                    [timeZone]
```

https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-odata/61934eae-311a-4af4-b8f8-82c112248651

Reply:
> Edm.DateTime - while odata.org has is just datetime’yyyy-mm-ddThh:mm[:ss[.fffffff]]’ - and SAP Gateway returns this in UTC by design

It will never include any timezone information, correct?

If so: Since Pythons datetime objects can be naive (explicitly not knowing anything about timezone) or aware (having some kind of timezone information), what about working with this to keep track whether a value is timezone aware or not instead of adding a fake UTC timezone?

> the MS ODATA specification for odata has optional timezone as well for this.

This refers to [version 3.0 of  OData](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-odata/c8490c34-adbc-43ee-a8c6-89760c123484). I think it should not be used as reference for OData 2.0.

> That's why I thought that what you did for Edm.DateTimeOffset type should be applied to Edm.DateTime as well - and apply UTC as the default, if no infomration about timezone is received.

Changing Edm.DateTime and Edm.DateTimeOffset so that the only change will be that the first one defaults to UTZ and the later one will fail if the offset/timezone information is missing should be easy to be implemented when /consuming/ data.

However, when /sending/ an aware (possibly non-UTC) Python datetime object to a server (converting it to its literal form) however, I can see two options:
1. Just expecting the server to deal with the timezone information. If there are implementations which can not handle it, hopefully someone will create a bug report.
2. "Normalize" the time information to UTZ. Implies that receiving entities, and then using them (unmodified) to "update" data on the server strips the previously existing timezone information. Feels wrong.

I think both attempts are not sound and Edm.DateTime should be strictly agnostic of timezones, while Edm.DateTimeOffset requires it.

In case the SAP Gateway never adds a timezone information to Edm.DateTime, this approach should work well with it.

Reply:
OK. I understand this long crossed comment as that the request for covering timezone support for Edm.DateTime is accepted. If it is true, thanks for that. I will leave this PR open.

I would like only to address the odata v3 part in the comment - yes, the MS documentation is quite a chaotic one. They version the document itself, not by the protocol version. The OASIS is clearer for understanding. However, IMHO the document is essentialy "everything is applied to both odata v2 and odata v3" - and things that are relevant to one protocol version only are listed  in the section 1.7.1 and 1.7.2 - https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-odata/31a103dd-acaa-4e3d-a637-2d5de28a20a1